src/services/completions.ts

```diff
@@ -1526,6 +1532,44 @@ namespace ts.Completions {
             });
         }
 
+        function findUndeclaredVariableIfExist(node: Node): VariableDeclaration | undefined {
+            if (!isIdentifier(node)) {
+                return undefined;
+            }
+
+            let curNode: Node = node.parent;
+            while (curNode && curNode.parent) {
+                if (isPropertyAssignment(curNode) && isObjectLiteralExpression(curNode.parent)) {
+                    curNode = curNode.parent.parent;
+
+                    if (isVariableDeclaration(curNode)) {
+                        return curNode;
+                    }
+                }
+                else {
+                    return undefined;
+                }
+            }
+
+            return undefined;

```

Comment:
Take a look at `findAncenstor` and see if that lets you write this more concisely.

Reply:
But I find that in this case, filterObjectMembersList will not be executed

Reply:
> Take a look at `findAncenstor` and see if that lets you write this more concisely.
I used findAncenstor at first, but I found that findAncenstor didn't do much of the work because I had to filter two nodes at a time, one for property assignments and one for object literal expression nodes